File: iterator.p.py
import time
import sys

def is_prime(n):
    for i in range(3, int(n / 2), 2):
        if n % i == 1:
            return False

    return True

max = int(sys.argv[len(sys.argv) - 1])

s = time.time()

prime = 0
for i in range(1, max, 2):
    if is_prime(i):
        prime += 1

print((time.time() - s) * 1000)

File: iterator.p.rs
use std::{env, time::Instant};

fn is_prime(n: i64) -> bool {
    for i in (3..n / 2).step_by(2) {
        if n % i == 0 {
            return false;
        }
    }

    return true;
}

fn main() {
    let max: i64 = env::args().last().map(|v| v.parse().unwrap()).unwrap();

    let start = Instant::now();
    let mut prime = 0;

    for i in (1..max).step_by(2) {
        if is_prime(i) {
            prime += 1;
        }
    }

    println!("{}", start.elapsed().as_secs_f64() * 1000.);
    drop(prime);
}
